river/neighbors/base.py

```diff
@@ -33,112 +34,11 @@ def __call__(self, a, b):
         return self.distance_function(a[0], b[0])
```

Comment:
I don't think we need this, right? Can't we just pass a regular function?

Reply:
I just don't have a better idea :/